traits/tests/test_new_notifiers.py

```diff
@@ -26,7 +27,7 @@ def test_notification_on_separate_thread(self):
         notifications = []
 
         def on_foo_notifications(obj, name, old, new):
-            thread_id = thread.get_ident()
+            thread_id = sm._thread.get_ident()
```

Comment:
Taking Python 3's hint that `_thread` should be considered private, and replacing this with `threading.current_thread().ident` (which should work out of the box on both Python 2 and Python 3). Same again below.